packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-trigger-template.ts

```diff
@@ -45,6 +46,13 @@ export class CustomResourceAuthStack extends cdk.Stack {
         type: 'String',
       });
       createPermissionToInvokeLambda(this, fnName, userpoolArn, config);
+      const permission = props.permissions.find(permission => config.triggerType === permission.trigger);
```

Comment:
Does `props.permissions` always have a value?

Reply:
Nice catch. Updated . Now this is checked in generateNestedAuthTriggerTemplate() Fn.